### ISSUE TITLE
Reorganize menus

### ISSUE THREAD

**Issue by jackburton79** [nexus6-haiku/Genio#42]
Menus should be reorganized. Currently they don't respect Haiku HIGs. Moreover the Project and File menus are a bit redundant.

**Comment by jackburton79** [comment]
I gave a shot at a (first) menu reorganization proposal.

1 File = no change
2 Edit = no change
3 Search = no change
4 Project
	-> Open (Project)
	-> Close (Project)
	-> Build (Project)
	-> Clean (Project)
	-> Build Mode
		-> Release
		-> Debug
	-> Make catkeys
        -> Make bindcatalogs
	-> Settings...
5 Run
	-> Run (target)
	-> Debug
6 Git = no change
7 Window
	-> Editor (or View)
		-> Zoom In
		-> Zoom Out
		-> Zoom Reset
	-> Interface = no change
	-> Settings...
8 Help = no change

Basically, some reorganization of top level menus, "Build" would go under "Project", "View" would go under "Window", and there would be a new "Run" top level menu.

"Run" in the future will contain all the run / debug stuff
"Project" all related project stuff

**Comment by Freaxed** [comment]
Consideration  regarding "Toggle line endings" and "Toggle white spaces"  under the Edit menu:
they should be moved under the 'Editor (or View)' submenu  near  the zoom items as they all affect the editor appeareance. The point is that the Zoom functions are affecting all the editors while the others are affecting only the current editor. I'm for making the two 'toggle' functions global as the zoom ones and also store the setting in the configuration. What do you think?

On another topic: Is there a better name for 'Interface'?

**Comment by jackburton79** [comment]
> Consideration regarding "Toggle line endings" and "Toggle white spaces" under the Edit menu: they should be moved under the 'Editor (or View)' submenu near the zoom items as they all affect the editor appeareance. The point is that the Zoom functions are affecting all the editors while the others are affecting only the current editor. I'm for making the two 'toggle' functions global as the zoom ones and also store the setting in the configuration. What do you think?

I agree on the "Toggle line endings" and "Toggle white spaces" menu items.
I'm not sure I understood tha latter part, though.

> 
> On another topic: Is there a better name for 'Interface'?

Appearance, maybe ?

**Comment by Freaxed** [comment]
> I agree on the "Toggle line endings" and "Toggle white spaces" menu items. I'm not sure I understood tha latter part, though.

if you select 'toggle the white space'  this will affect only the current open editor, if you open a new file or switch to another tab you don't see any change. If you change the Zoom this is affecting all the editor (opened or that will be open). Morehover, the zoom is stored in the config settings while the "toggles" are not.

> Appearance, maybe ?

Better!

**Comment by nexus6-haiku** [comment]
1. I would propose to put the Run menu items under Project as they are actions performed on a Project, after all.
Having a separate menu would anticipate the possibility to debug within Genio (Step in, Step out, etc.).
Although it would be cool, this will not happen anytime soon, I think.

2. All the general settings and items affecting the UI should go under one menu while Editor settings may go under the Edit menu, maybe separated by a custom separator?
What about the first menu with the app Icon, like in the old BeOS days?
I think a custom menu would be required but I've seen apps around sporting it.

**Comment by jackburton79** [comment]
> 1. I would propose to put the Run menu items under Project as they are actions performed on a Project, after all.
> Having a separate menu would anticipate the possibility to debug within Genio (Step in, Step out, etc.).
> Although it would be cool, this will not happen anytime soon, I think.

That's a good point.

> 2. All the general settings and items affecting the UI should go under one menu while Editor settings may go under the Edit menu, maybe separated by a custom separator?

I think Edit should contain actions which act on the editing aspect, while actions which act on the appearance / UI should go inside Window, maybe in a submenu

> What about the first menu with the app Icon, like in the old BeOS days?
> I think a custom menu would be required but I've seen apps around sporting it.

IIRC there is a private class available in the haiku tree.

**Comment by jackburton79** [comment]
I started changing menus as we discussed here.
I did not move View menu yet.

**Comment by jackburton79** [comment]
Partially done. For now it'll do.
